scripts/compare_wandb_configs.py

```diff
@@ -24,6 +38,78 @@ def parse_run_path(run_path: str) -> str:
     raise ValueError(f"Could not parse '{run_path}'")
 
 
+def print_keys_with_differences(left_config, right_config, level=0):
+    prefix = "\t\t" * level
+
+    s_left = ""
+    left_only_keys = left_config.keys() - right_config.keys()
+    if len(left_only_keys) > 0:
+        s_left += prefix + "Settings only in left:\n"
+        s_left += (prefix + "\n").join(f"\t{k}: {left_config[k]}" for k in sorted(left_only_keys)) + "\n"
+
+    s_right = ""
+    right_only_keys = right_config.keys() - left_config.keys()
+    if len(right_only_keys) > 0:
+        s_right += prefix + "Settings only in right:\n"
+        s_right += (prefix + "\n").join(f"\t{k}: {right_config[k]}" for k in sorted(right_only_keys)) + "\n"
+
+    s_shared = ""
+    keys_with_differences = {
+        k for k in left_config.keys() & right_config.keys() if left_config[k] != right_config[k]
+    }
+    if len(keys_with_differences) > 0:
+        for k in sorted(keys_with_differences):
+            if isinstance(left_config[k], list) and isinstance(right_config[k], list):
+                s_list = prefix + f"{k}:\n"
+                for left, right in zip(left_config[k], right_config[k]):  # assumes lists are same order
+                    if isinstance(left, dict) and isinstance(right, dict):
+                        print_keys_with_differences(left_config=left, right_config=right, level=level + 1)
+                    else:
+                        s_list += prefix + f"\t{left}\n" + prefix + f"\t{right}\n\n"
+                if s_list != prefix + f"{k}:\n":
+                    s_shared += s_list
+            else:
+                s_shared += prefix + f"{k}\n\t{left_config[k]}\n" + prefix + f"\t{right_config[k]}\n\n"
+
+    if (s_left or s_right) and not s_shared:
+        s = s_left + "=" * 50 + "\n" + s_right + "=" * 50 + "\n" + prefix + "No differences in shared settings.\n"
+    else:
+        s = s_left + "=" * 50 + "\n" + s_right + "=" * 50 + "\n" + s_shared
+    print(s.strip())
+    return
+
+
+def simplify_path(path):
+    # Remove the variable prefix up to and including 'preprocessed/'
+    path = re.sub(r"^.*?preprocessed/", "", path)
+
+    # Split the path into components
+    components = path.split("/")
+
+    # Remove common suffixes like 'allenai'
+    components = [c for c in components if c != "allenai"]
```

Comment:
You don't think that's confusing? For copy and paste?

Reply:
removed